common/core/gpudevice.cpp

```diff
@@ -103,14 +117,36 @@ void GpuDevice::GetConnectedPhysicalDisplays(
 }
 
 bool GpuDevice::EnableDRMCommit(bool enable, uint32_t display_id) {
-  // TODO clean all display for commit status
   size_t size = total_displays_.size();
   bool ret = false;
   if (size > display_id)
     ret = total_displays_.at(display_id)->EnableDRMCommit(enable);
   return ret;
 }
 
+bool GpuDevice::ResetDrmMaster(bool drop_master) {
+  // clean all display for commit status
+  bool ret = true;
+  if (drop_master) {
+    ResetAllDisplayCommit(false);
+    display_manager_->DropDrmMaster();
+    ITRACE("locking %s and monitoring if %s is unlocked.", HWC_LOCK_FILE,
+           HWC_LOCK_FILE);
+    // launch a thread to check hwc.lock and re-apply drm master.
```

Comment:
You don't need a new thread
HWC has already a thread for vsync
In the vsync thread, you can add logic

Check HWC is master or not,
If not:
If( no hwc.lock or it is not locked) {
setSelftoBeDRM master
ForeceRefresh(); //this will force SurfaceFlinger to send Present Call to HWC, so the UI appear immediately
}

Reply:
The GpuDevice Thread is original here for checking hwc.lock. we just resume it here. 
In original implementation, this thread will exit after getting the lock and commiting frame.
Now, the thread will be waiting after getting the lock and commiting frame in booting first run.
and be resumed once this API is invoked by kmscube/earlyevs.

Reply:
I will add the logic you mentioned into vblank thread.